app/src/main/java/org/simple/clinic/summary/addphone/AddPhoneNumberDialog.kt

```diff
@@ -0,0 +1,126 @@
+package org.simple.clinic.summary.addphone
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.widget.Button
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.FragmentManager
+import com.google.android.material.textfield.TextInputLayout
+import com.jakewharton.rxbinding2.view.RxView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import kotterknife.bindView
+import org.simple.clinic.R
+import org.simple.clinic.activity.TheActivity
+import org.simple.clinic.patient.PatientUuid
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.UiEvent
+import org.simple.clinic.widgets.showKeyboard
+import javax.inject.Inject
+
+class AddPhoneNumberDialog : AppCompatDialogFragment() {
+
+  companion object {
+    private const val FRAGMENT_TAG = "AddPhoneNumberDialog"
+    private const val KEY_PATIENT_UUID = "patientUuid"
+
+    fun show(patientUuid: PatientUuid, fragmentManager: FragmentManager) {
+      val existingFragment = fragmentManager.findFragmentByTag(FRAGMENT_TAG)
+
+      if (existingFragment != null) {
+        fragmentManager
+            .beginTransaction()
+            .remove(existingFragment)
+            .commitNowAllowingStateLoss()
+      }
+
+      val fragment = AddPhoneNumberDialog().apply {
+        isCancelable = false
+        arguments = Bundle(1).apply {
+          putSerializable(KEY_PATIENT_UUID, patientUuid)
+        }
+      }
+
+      fragmentManager
+          .beginTransaction()
+          .add(fragment, FRAGMENT_TAG)
+          .commitNowAllowingStateLoss()
+    }
+  }
+
+  @Inject
+  lateinit var controller: AddPhoneNumberDialogController
+
+  private val phoneInputLayout by bindView<TextInputLayout>(R.id.updatephone_phone_inputlayout)
+  private val numberEditText by bindView<EditText>(R.id.updatephone_phone)
+
+  private val onStarts = PublishSubject.create<Any>()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    TheActivity.component.inject(this)
+  }
+
+  @SuppressLint("CheckResult", "InflateParams")
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val layout = LayoutInflater.from(requireContext()).inflate(R.layout.dialog_patientsummary_updatephone, null)
```

Comment:
The message on the dialog reads "Patient's phone number was previously marked as incorrect" when it should be "Patient's phone number was not entered during registration".

Reply:
Whoops, fixing this

Reply:
I had already fixed this, but I think I lost the commit while trying to recover changes from `git reflog`.